hal/network/ncp_client/quectel/quectel_ncp_client.cpp

```diff
@@ -1397,6 +1404,43 @@ int QuectelNcpClient::setModuleFunctionality(CellularFunctionality cfun, bool ch
     return r;
 }
 
+int QuectelNcpClient::getPolicymanServiceMode() {
+    auto resp = parser_.sendCommand(QUECTEL_CFUN_TIMEOUT, "AT+QNVFR=\"policyman/svc_mode\"");
+    int curVal = (int)CellularPolicymanServiceMode::NONE;
+    auto r = resp.scanf("+QNVFR: %d", &curVal);
+    // Soft response on ERROR
+    const int result = CHECK_PARSER(resp.readResult());
+    if (result != AtResponse::OK) {
+        return (int)CellularPolicymanServiceMode::NONE;
+    }
+    CHECK_TRUE(r == 1, (int)CellularPolicymanServiceMode::NONE);
+    return curVal;
+}
+
+int QuectelNcpClient::setPolicymanServiceMode(CellularPolicymanServiceMode mode, bool check) {
+    int curVal = (int)CellularPolicymanServiceMode::NONE;
+    if (check) {
+        curVal = getPolicymanServiceMode();
+        if ((int)mode == curVal || curVal == (int)CellularPolicymanServiceMode::NONE) {
+            // Already in required state, or command error'd/not supported.
+            return 0;
+        }
+    }
+
+    int r = SYSTEM_ERROR_UNKNOWN;
+
+    if (mode == CellularPolicymanServiceMode::FULL_SERVICE) {
+        r = parser_.execCommand(10000, "AT+QNVFW=\"policyman/svc_mode\",02");
+    } else if (mode == CellularPolicymanServiceMode::NO_SERVICE) {
+        r = parser_.execCommand(10000, "AT+QNVFW=\"policyman/svc_mode\",00");
+    }
+
+    CHECK_PARSER(r);
```

Comment:
Maybe add a warning log if we don't get an OK back? But I guess we will anyway see that in the AT log..